www/include/monitoring/status/Hosts/xml/hostXML.php

```diff
@@ -192,66 +199,83 @@
 }
 
 if ($hostgroups) {
-    $rq1 .= " AND h.host_id = hhg.host_id " .
-        "AND hg.hostgroup_id IN (" . $hostgroups . ") " .
-        "AND hhg.hostgroup_id = hg.hostgroup_id";
+    $rq1 .= " AND h.host_id = hhg.host_id
+        AND hg.hostgroup_id IN (:hostgroup)
+        AND hhg.hostgroup_id = hg.hostgroup_id";
+    $queryValues['hostgroup'] = [
+        \PDO::PARAM_INT => $hostgroups
+    ];
 }
 
 if ($instance != -1 && !empty($instance)) {
-    $rq1 .= " AND h.instance_id = " . $instance;
+    $rq1 .= " AND h.instance_id = :instance ";
+    $queryValues['instance'] = [
+        \PDO::PARAM_INT => $instance
+    ];
 }
-$rq1 .= " AND h.enabled = 1 ";
+$rq1 .= " AND h.enabled = 1";
 switch ($sort_type) {
     case 'name':
         $rq1 .= " ORDER BY h.name " . $order;
         break;
     case 'current_state':
-        $rq1 .= " ORDER BY h.state " . $order . ",h.name ";
+        $rq1 .= " ORDER BY h.state " . $order . ",h.name";
         break;
     case 'last_state_change':
-        $rq1 .= " ORDER BY h.last_state_change " . $order . ",h.name ";
+        $rq1 .= " ORDER BY h.last_state_change " . $order . ",h.name";
         break;
     case 'last_hard_state_change':
-        $rq1 .= " ORDER BY h.last_hard_state_change " . $order . ",h.name ";
+        $rq1 .= " ORDER BY h.last_hard_state_change " . $order . ",h.name";
         break;
     case 'last_check':
-        $rq1 .= " ORDER BY h.last_check " . $order . ",h.name ";
+        $rq1 .= " ORDER BY h.last_check " . $order . ",h.name";
         break;
     case 'current_check_attempt':
-        $rq1 .= " ORDER BY h.check_attempt " . $order . ",h.name ";
+        $rq1 .= " ORDER BY h.check_attempt " . $order . ",h.name";
         break;
     case 'ip':
-        # Not SQL portable
-        $rq1 .= " ORDER BY IFNULL(inet_aton(h.address), h.address) " . $order . ",h.name ";
+        // Not SQL portable
+        $rq1 .= " ORDER BY IFNULL(inet_aton(h.address), h.address) " . $order . ",h.name";
         break;
     case 'plugin_output':
-        $rq1 .= " ORDER BY h.output " . $order . ",h.name ";
+        $rq1 .= " ORDER BY h.output " . $order . ",h.name";
         break;
     case 'criticality_id':
```

Comment:
as 'criticality_id' and default case will add the same partial string to the request, we let it empty